### PR TITLE
added LibraryMenuTest, BookmarksRobot, HistoryRobot

### PR DESCRIPTION
**Pull request by sv-ohorvath**
Added LibraryMenuTest, BookmarksRobot, HistoryRobot

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by sv-ohorvath** [comment]
All tests passed on Firebase: https://console.firebase.google.com/u/2/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/4943915023647204977

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/HistoryRobot.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    class Transition{}
+}
+
+fun historyMenu(interact: HistoryRobot.() -> Unit): HistoryRobot.Transition {
+    HistoryRobot().interact()
+    return HistoryRobot.Transition()
+}
+
+private fun assertHistoryView() {
+    // verify that we are in the correct history view
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/HistoryRobot.kt]: the function name is adequate so, no comment necessary

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/LibraryRobot.kt [hunk 2]
```diff
@@ -34,13 +39,57 @@ class LibraryRobot {
             HomeScreenRobot().interact()
             return HomeScreenRobot.Transition()
         }
+
+        fun closeMenu(interact: HomeScreenRobot.() -> Unit): HomeScreenRobot.Transition {
+            mDevice.waitForIdle()
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/LibraryRobot.kt]: see:  https://github.com/mozilla-mobile/fenix/pull/4770/files#r316397535

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/LibraryRobot.kt [hunk 3]
```diff
@@ -34,13 +39,57 @@ class LibraryRobot {
...
+            closeButton().click()
+
+            HomeScreenRobot().interact()
+            return HomeScreenRobot.Transition()
+        }
+
+        fun openBookmarks(interact: BookmarksRobot.() -> Unit): BookmarksRobot.Transition{
+            mDevice.waitForIdle()
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/LibraryRobot.kt]: same, see:  https://github.com/mozilla-mobile/fenix/pull/4770/files#r316397535

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/LibraryRobot.kt [hunk 4]
```diff
@@ -34,13 +39,57 @@ class LibraryRobot {
...
+            bookmarksButton().perform(click())
+
+            BookmarksRobot().interact()
+            return BookmarksRobot.Transition()
+        }
+
+        fun openHistory(interact: HistoryRobot.() -> Unit): HistoryRobot.Transition{
+            mDevice.waitForIdle()
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/LibraryRobot.kt]: same, see:  https://github.com/mozilla-mobile/fenix/pull/4770/files#r316397535
but only add these waits if necessary (i.e. if test fails with an element not found)

File: app/src/androidTest/java/org/mozilla/fenix/ui/LibraryMenuTest.kt [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+class LibraryMenuTest {
+
+    @get:Rule
+    val activityTestRule = HomeActivityTestRule()
+
+    @Test
+    fun libraryMenuItemsTest() {
+        homeScreen {
+        }.openThreeDotMenu { }
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/LibraryMenuTest.kt]: the desired syntax for transitions is slightly different....
```
}.openThreeDotMenu {
}.openLibrary {
```
The brackets around transition functions are always on the same line and always face outward for readability.
(I know, it looks really weird 🤷‍♂ )

File: app/src/androidTest/java/org/mozilla/fenix/ui/LibraryMenuTest.kt [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            .openLibrary {
+                verifyLibraryView()
+                verifyHistoryButton()
+                verifyBookmarksButton()
+            }
+    }
+
+    @Test
+    fun closeMenuButtonTest() {
+        homeScreen {
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/LibraryMenuTest.kt]: yea, these transitions look good 👍

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BookmarksRobot.kt [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+import androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import org.hamcrest.Matchers.allOf
+import org.mozilla.fenix.R
+
+class BookmarksRobot {
+
+    fun verifyBookmarksMenuView() = assertBookmarksView()
+
+    class Transition
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/BookmarksRobot.kt]: It looks like this empty Transition class should be removed (not doing anything).  Let me know if you have issues with this.

Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/BookmarksRobot.kt]: This is used in LibraryRobot's openBookmarks() Transition method and then in bookmarksButtonTest().

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/LibraryRobot.kt (hunk 3 above)
Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/LibraryRobot.kt]: the tests passed without it, so I removed both waits from openBookmarks() and openHistory()

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BookmarksRobot.kt (hunk 7 above)
Comment (rpappalax) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/BookmarksRobot.kt]: 🤔 hmmm, I'm not sure it's a good practice to leave an empty transition class.  If the tests works with an empty transition class but fails when you remove it, perhaps we just need to rewrite the methods.

**Comment by rpappalax** [comment]
👍 great job Oana

**Pull request merged**